### DISCUSSION TITLE
Variance Importance Plot question

### DISCUSSION THREAD

**Discussion by MiguelFudolig** [kogalur/randomForestSRC#429] [Q&A] [answered]
Hello, what if I want to just plot the "all" column of the variable importance plot for an imbalanced dataset? I'm trying to replicate the output in the website but the code yields extra values in the y-axis as shown.

![image](https://github.com/kogalur/randomForestSRC/assets/71338141/d80ee388-a7b4-41a9-95ce-4d53ca638ca5)

Thanks,
M

**Comment by ishwaran** [comment, marked as answer]
The importance matrix is stored in the grow object: extract it and you can create any plot you want.

Here's an example for the iris data:


```
library(ggplot2)

## rf-classification
rf_model <- rfsrc(Species~., iris, importance = "permute")

# Extract the "all" variable importance values 
var_importance <- rf_model$importance[, "all"]
var_importance_df <- data.frame(variable = names(var_importance), importance = var_importance)
  
# Create ggplot for variable importance
importance_plot <- ggplot(var_importance_df, aes(x = reorder(variable, importance), y = importance)) +
  geom_bar(stat = "identity", fill = "#F8766D") +
  coord_flip() +
  labs(title = "Variable Importance", x = "Variable", y = "Importance") +
  theme_minimal()
  
plot(importance_plot)
```
